src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/DbBatch.IAsyncDispose.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.Common
+{
+    public partial class DbBatch : IAsyncDisposable
```

Comment:
Since we are only implementing the new batch APIs in netcore, can this be internal for netfx use?

Reply:
The definition of DbBatch has to be public for us to expose another class which derives from it. It's only public and present in the src not in the ref. So it's not language internal but it's as hidden as we can get it.